cypress-setup.sh

```diff
@@ -14,19 +14,27 @@ else
   _PRISMIC_REPO=$REPOSITORY
 fi
 
-if [[ "$PRISMIC_URL" == 'https://wroom-qa.com' ]] || [[ "$PRISMIC_URL" == 'http://wroom.test' ]] ; then 
-  MODE="dev"
+if [[ "$PRISMIC_URL" == 'https://wroom-qa.com' ]] || [[ "$PRISMIC_URL" == 'http://wroom.test' ]] ; then
+  export SM_ENV="development"
 elif [[ "$PRISMIC_URL" == 'https://wroom.io' ]]; then
-  MODE="stage"
+  export SM_ENV="staging"
 else
-  MODE="prod"
+  export SM_ENV="production"
 fi
 
+# TODO: Let adapters decalre their dependencies and automatically install
+# during `@slicemachine/init`. The fix may be as simple as declaring them as
+# peer dependencies. Line 37 should be removed once that functionality is
+# added.
+
 rm -rf e2e-projects/cypress-next-app \
 && npx --yes create-next-app e2e-projects/cypress-next-app \
-&& npx --yes ts-node ./cypress/plugins/addAuth.ts ${EMAIL} ${PASSWORD} ${PRISMIC_URL} \
-&& npx --yes ts-node ./cypress/plugins/createRepo.ts "${_PRISMIC_REPO}" "${PASSWORD}" "${PRISMIC_URL}"  \
+&& npx --yes vite-node ./cypress/plugins/addAuth.ts -- ${EMAIL} ${PASSWORD} ${PRISMIC_URL} \
+&& npx --yes vite-node ./cypress/plugins/createRepo.ts -- "${_PRISMIC_REPO}" "${PASSWORD}" "${PRISMIC_URL}"  \
+&& npm --workspaces pack --pack-destination ./e2e-projects/cypress-next-app \
 && cd e2e-projects/cypress-next-app \
-&& node ../../packages/init/build/index.js --mode "${MODE}" --repository "${_PRISMIC_REPO}" \
-&& npm i \
-&& npx --yes json -I -f package.json -e "this.scripts.slicemachine=\"node ../../packages/slice-machine/bin/start.js\""
+&& npm i *.tgz \
```

Comment:
what's `npm i *.tgz` this for? 
will there be tarballs laying around somewhere?

Reply:
In order for the Cypress tests to run using the latest development versions of Slice Machine and its subpackages, we use `npm pack` to package each subpackage, then install them into the temporary Cypress project using `npm i *.tgz`.

There will be tarball files in each package directory, but they are in `.gitignore` and have clear filenames.

We could clean up the `.tgz` files after installation, but we would need to worry about cross-OS compatibility (`rm` won't work here).

---

How do the current Cypress tests run using the unpublished versions of SM?

Reply:
Currently cypress uses the local build and the package.json gets modified to use the correct one, it's a bit hacky and I've suggested a few times to use `verdaccio` to have. a local npm repository to publish too, but no-ones had the time to implement it